code/core/src/core-server/utils/open-browser/opener.test.ts

```diff
@@ -0,0 +1,117 @@
+import type { ChildProcess } from 'node:child_process';
+
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
+
+import spawn from 'cross-spawn';
+import open from 'open';
+
+import { openBrowser } from './opener';
+
+vi.mock('open', { spy: true });
+vi.mock('cross-spawn', { spy: true });
+
+describe('openBrowser BROWSER script handling', () => {
+  const originalEnv = { ...process.env };
+  const originalArgv = [...process.argv];
+  let platformSpy: ReturnType<typeof vi.spyOn>;
+
+  beforeEach(() => {
+    vi.resetAllMocks();
+    platformSpy = vi.spyOn(process, 'platform', 'get').mockReturnValue('linux');
+    process.env = { ...originalEnv };
+    process.argv = ['node', 'test'];
+
+    vi.mocked(open).mockImplementation(() => {
+      return Promise.resolve({} as unknown as Awaited<ReturnType<typeof open>>);
+    });
+    const child = { on: vi.fn() } as unknown as ChildProcess;
+    vi.mocked(spawn).mockImplementation(() => child);
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+    process.env = originalEnv;
+    process.argv = originalArgv;
+  });
+
+  it('executes a node script when BROWSER points to a JS file', () => {
+    process.env.BROWSER = '/tmp/browser.js';
```

Comment:
@ndelangen do we have an env mocking library already? I've used `mockedEnv` in the past. I'm concerned we're not cleaning up `env` here so this could lead to future tests on the same runner behaving differently.